apps/wallet/src/background/keyring/storage-utils.ts

```diff
@@ -0,0 +1,9 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import type { Storage } from 'webextension-polyfill';
+
+export const SESSION_STORAGE: Storage.LocalStorageArea | null =
+    // @ts-expect-error chrome
+    global?.chrome?.storage?.session || null;
+export const IS_SESSION_STORAGE_SUPPORTED = !!SESSION_STORAGE;
```

Comment:
Another option here: Rather than doing this checking you could just export a `set` and `get` function here, and detect `session` inline instead. In the event that it's not supported, you just store it in something in-memory.